src/Shape.cpp

```diff
@@ -22,6 +22,80 @@ bool Shape::intersect(const Ray &, float t0, float t1, double& distance) const {
     return false;
 }
 
+
+/** @brief Shape::intersect() determines whether a point p lies within the shape.
+ *  @return bool True means point p lies inside the shape.
+ *  @math Let the line segment P connect points p and an arbitrary point p_out outside of the shape
+ *  We count the number of intersections between P and the shape. A positive number means point p is inside the shape.
+ *  We use plucker coordinates to determine whether or not a triangle intersects line segment P
+ **/
+
+bool Shape::intersect(const Vector3& p) const {
```

Comment:
To me, intersection means "get the common part". I can also understand line intersection as in "crossing", but your description decides "point_is_inside". It uses line intersection (probably), but a point and a shape don't intersect (cross) imho, it's either "in" or "out".

Reply:
I agree, this terminology was copied from box.cpp. But a name like contains() would make more sense